lib/puppet/face/strings.rb

```diff
@@ -81,6 +81,65 @@
     end
   end
 
+  action(:describe) do #This is Kris' experiment with string based describe
+    option "--list" do
+      summary "list types"
+    end
+    option "--providers" do
+      summary "provide details on providers"
+    end
+
+#TODO: Implement the rest of describe behavior
+#     * --help:
+#   Print this help text
+
+# * --providers:
+#   Describe providers in detail for each type
+
+# * --list:
+#   List all types
+
+# * --meta:
+#   List all metaparameters
+
+# * --short:
+#   List only parameters without detail
+
+
+    when_invoked do |*args|
+      check_required_features
+      require 'puppet-strings'
+
+      options = args.last
+      options[:describe] = true
+      options[:stdout] = true
+      options[:format] = 'json'
+      
+      if args.length > 1
+        if options[:list]
+          $stderr.puts "WARNING: ignoring types when listing all types."
+        else
+          options[:describe_types] = args[0..-2]
+        end
+      end
+
+      #TODO: Set up search_patterns and whatever else needed to collect data for describe - currently missing some
+      #          manifests/**/*.pp
+      #          functions/**/*.pp
+      #          tasks/*.json
+      #          plans/*.pp
+      search_patterns = %w(
+        types/**/*.pp
+        lib/**/*.rb
+      )
+      PuppetStrings::generate(
```

Comment:
I wonder about trying to create a `PuppetStrings::describe` method separate from `PuppetStrings::generate`. We may be able to cut some of the cruft out to reduce load time.

Reply:
Any suggestions for stuff we don't need for describe @eputnam ?

Reply:
I forget now what I was thinking here. I guess that describe doesn't need to look at everything we list in generate normally? But as for what that is, I can't remember.

Reply:
Yeah, I guess the search_patterns I set up here are much less than the regular generate.  I think for now I will focus on filling in missing data, then I will come back and look for performance improvements.